packages/core/src/ceramic.ts

```diff
@@ -634,6 +634,13 @@ export class Ceramic implements CeramicApi {
       let stream
       try {
         if (query.genesis) {
+          if (
+            StreamUtils.isSignedCommitContainer(query.genesis) ||
+            StreamUtils.isSignedCommit(query.genesis)
+          ) {
+            return Promise.reject('Given genesis commit is not deterministic')
+          }
+
           const genesisCID = await this.ipfs.dag.put(query.genesis)
           if (!streamId.cid.equals(genesisCID)) {
             return Promise.reject('Given StreamID CID does not match given genesis content')
```

Comment:
```suggestion
            throw new Error('Given StreamID CID does not match given genesis content')
```

Reply:
I've been able to test this approach using the following:

```
expect(
        ceramic2.multiQuery([
          {
            streamId: stream1.id,
            genesis: genesisCommit,
          },
        ])
      ).rejects.toEqual('Given StreamID CID does not match given genesis content')
      ```
      
I'm not sure how I would test the error throw as the overall function does not care about errors being thrown within `walkNext`

Reply:
ah I see. You're trying to make failures of this specific type actually cause the whole operation to fail, unlike the existing failures when loading one of the linked streams, which should still allow the rest of the operation to continue.  Instead of using a different error reporting technique to get around the try/catch, why not just move the block of code you added to before the start of the try/catch block?